vos/vos/vos.py

```diff
@@ -2952,3 +2847,215 @@ def __iter__(self):
+            for node in protocol.findall(Node.ENDPOINT):
+                result.append(node.text)
+        return result
+
+    def _get_phase(self, phase_url):
+        response = self.endpoints.session.get(phase_url,
+                                              allow_redirects=True)
+        response.raise_for_status()
+        return response.text
+
+    def get_transfer_error(self, url, uri):
+        """Follow a transfer URL to the Error message
+        :param url: The URL of the transfer request that had the error.
+        :param uri: The uri that we were trying to transfer (get or put).
+
+        :raises When a network problem occurs, it raises one of the
+        HttpException exceptions declared in the
+        cadcutils.exceptions module
+        """
+        error_codes = {'NodeNotFound': errno.ENOENT,
+                       'RequestEntityTooLarge': errno.E2BIG,
+                       'PermissionDenied': errno.EACCES,
+                       'OperationNotSupported': errno.EOPNOTSUPP,
+                       'InternalFault': errno.EFAULT,
+                       'ProtocolNotSupported': errno.EPFNOSUPPORT,
+                       'ViewNotSupported': errno.ENOSYS,
+                       'InvalidArgument': errno.EINVAL,
+                       'InvalidURI': errno.EFAULT,
+                       'TransferFailed': errno.EIO,
+                       'DuplicateNode.': errno.EEXIST,
+                       'NodeLocked': errno.EPERM}
+        job_url = str.replace(url, "/results/transferDetails", "")
+
+        try:
+            phase_url = job_url + "/phase"
+            sleep_time = 1
+            roller = ('\\', '-', '/', '|', '\\', '-', '/', '|')
+            phase = self._get_phase(phase_url)
+            # do not remove the line below. It is used for testing
+            logging.debug("Job URL: " + job_url + "/phase")
+            while phase in ['PENDING', 'QUEUED', 'EXECUTING', 'UNKNOWN']:
+                # poll the job. Sleeping time in between polls is doubling
+                # each time until it gets to 32sec
+                total_time_slept = 0
+                if sleep_time <= 32:
+                    sleep_time *= 2
+                slept = 0
+                if logger.getEffectiveLevel() == logging.INFO:
+                    while slept < sleep_time:
+                        sys.stdout.write(
+                            "\r%s %s" % (phase, roller[total_time_slept %
+                                                       len(roller)]))
+                        sys.stdout.flush()
+                        slept += 1
+                        total_time_slept += 1
+                        time.sleep(1)
+                    sys.stdout.write("\r                    \n")
+                else:
+                    time.sleep(sleep_time)
+                phase = self._get_phase(phase_url)
+                logging.debug(
+                    "Async transfer Phase for url %s: %s " % (url, phase))
```

Comment:
There is a mix of string format templates (i.e. `"%s" % value` vs `"{0}".format(value)`).  Please make these consistent to which ever method falls into current standards.

Reply:
Fixed